spec/job/adapters/linux_host/launcher_spec.rb

```diff
@@ -105,6 +105,28 @@ def content
                 subject.start_remote_session(build_script(native: {'submit_host_override' => alt_submit_host}))
             end
         end
+
+        context "When tmux returns extra data along with the hostname it" do
+            let(:broken_data) { 
+                <<~HEREDOC
+                Code Server (launched-by-ondemand-74542f25-fa6e-4e33-8e47-
+                feb166af57a9@---------------- ----------------------------
+                -------- Usage Statistics for project PZS0714 Time 2021-09-16 to
+                 2021-09-17 PI alanc@osc.edu Remaining Budget -0.25 
+                 ---------------- ------------------------------------ 
+                 User Jobs Dollars Status ------------- ------ --------- -------- 
+                 alanc 0 0.0 ACTIVE johrstrom 15 0.0 ACTIVE johrstromtest 0 0.0 ACTIVE 
+                 travert 9 0.0 ACTIVE -- -- -- TOTAL 24 0.0 owens-login01.hpc.osc.edu)
+             HEREDOC
+            }
+            it "parses remote host correctly" do
+                allow(Open3).to receive(:capture3).and_return([broken_data, '', exit_success])
+
+                expect{
+                    subject.start_remote_session(build_script).to match(/owens-login01\.hpc\.osc\.edu/)
+                }
```

Comment:
You have this entire thing wrapped in the expect block.

I think you want something more like this:
```ruby
session = subject.start_remote_session(build_script)
expect(session).to match(/^launched-by-ondemand.+@owens-login01\.hpc\.osc\.edu$/)
```

Indeed when you change the test case you'll see that that `parse_hostname` isn't working right.